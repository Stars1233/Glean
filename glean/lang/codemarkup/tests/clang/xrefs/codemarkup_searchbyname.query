query: codemarkup.search.SearchByName { searchcase = Sensitive, language = Cpp }
transform: [gensort, []]
