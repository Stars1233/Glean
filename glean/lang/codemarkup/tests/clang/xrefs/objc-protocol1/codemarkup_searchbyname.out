[
  "@generated",
  {
    "key": {
      "searchcase": 0,
      "name": "J",
      "entity": {
        "cxx": {
          "defn": {
            "objcContainer": {
              "key": {
                "declaration": {
                  "key": {
                    "id": { "interface_": { "key": "J" } },
                    "source": {
                      "file": { "key": "test.m" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 11,
                      "columnEnd": 4
                    }
                  }
                },
                "protocols": [ ],
                "members": {
                  "key": [
                    {
                      "objcMethod": {
                        "key": {
                          "selector": { "key": [ "checkProtocol" ] },
                          "locations": [
                            {
                              "file": { "key": "test.m" },
                              "span": { "start": 242, "length": 13 }
                            }
                          ],
                          "container": { "interface_": { "key": "J" } },
                          "signature": {
                            "key": {
                              "returns": { "key": "void" },
                              "parameters": [
                                {
                                  "name": { "key": "f" },
                                  "type": { "key": "Protocol *" }
                                }
                              ]
                            }
                          },
                          "isInstance": true,
                          "isOptional": false,
                          "isAccessor": false,
                          "source": {
                            "file": { "key": "test.m" },
                            "lineBegin": 10,
                            "columnBegin": 1,
                            "lineEnd": 10,
                            "columnEnd": 36
                          }
                        }
                      }
                    }
                  ]
                }
              }
            }
          }
        }
      },
      "location": {
        "name": "J",
        "file": { "key": "test.m" },
        "location": {
          "range": {
            "file": { "key": "test.m" },
            "lineBegin": 9,
            "columnBegin": 1,
            "lineEnd": 11,
            "columnEnd": 4
          }
        },
        "destination": {
          "file": { "key": "test.m" },
          "span": { "start": 232, "length": 1 }
        }
      },
      "kind": 12,
      "language": 0
    }
  },
  {
    "key": {
      "searchcase": 0,
      "name": "call",
      "entity": {
        "cxx": {
          "defn": {
            "function_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": { "name": { "key": "call" } } },
                        "scope": { "global_": { } }
                      }
                    },
                    "signature": {
                      "key": {
                        "returns": { "key": "void" },
                        "parameters": [
                          { "name": { "key": "s" }, "type": { "key": "J *" } }
                        ]
                      }
                    },
                    "source": {
                      "file": { "key": "test.m" },
                      "lineBegin": 16,
                      "columnBegin": 1,
                      "lineEnd": 19,
                      "columnEnd": 1
                    }
                  }
                },
                "isInline": false
              }
            }
          }
        }
      },
      "location": {
        "name": "call",
        "file": { "key": "test.m" },
        "location": {
          "range": {
            "file": { "key": "test.m" },
            "lineBegin": 16,
            "columnBegin": 1,
            "lineEnd": 19,
            "columnEnd": 1
          }
        },
        "destination": {
          "file": { "key": "test.m" },
          "span": { "start": 302, "length": 4 }
        }
      },
      "kind": 13,
      "language": 0
    }
  }
]